docker/Dockerfile

```diff
@@ -15,20 +15,46 @@ RUN apt-get update \
  && curl -sSL https://get.haskellstack.org/ | sh \
  && rm -rf /var/lib/apt/lists/*
 
+# COPY from https://github.com/commercialhaskell/stack/blob/master/etc/dockerfiles/arm64.Dockerfile#L3-L5
+FROM ubuntu:20.04 AS stack-build-env-arm64
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && apt-get install -y \
+        curl build-essential curl libffi-dev libffi7 libgmp-dev libgmp10 libncurses-dev libncurses5 libtinfo5 libnuma-dev xz-utils \
+        g++ gcc libc6-dev libgmp-dev make zlib1g-dev git gnupg netbase
+
+
+WORKDIR /tmp
+
+ENV LLVM_VERSION=9.0.1
+
+RUN curl -L https://github.com/llvm/llvm-project/releases/download/llvmorg-${LLVM_VERSION}/clang+llvm-${LLVM_VERSION}-aarch64-linux-gnu.tar.xz --output /tmp/llvm.tar.xz && \
+    unxz /tmp/llvm.tar.xz && \
+    tar xfv /tmp/llvm.tar --strip-components 1 -C /usr && \
+    rm /tmp/llvm.tar
+
+RUN curl -sSL https://github.com/commercialhaskell/stack/releases/download/v2.7.1/stack-2.7.1-linux-$(uname -m).bin > /usr/local/bin/stack && \
+    chmod +x /usr/local/bin/stack
+
+FROM stack-build-env-${TARGETARCH} AS builder
+
+RUN stack setup ghc-8.10.4
```

Comment:
I think this should be taken from a stack.yaml file instead (i.e. just call stack setup)

Reply:
Sure. i will clean it. 

It is an example of arm64 build env from stack repo. 😂

I haven't use Haskell and toolchains before.

Reply:
thanks again for adventuring!